packages/plugins/response-cache/src/index.ts

```diff
@@ -175,28 +176,42 @@ export default function useMeshResponseCache(
   options: YamlConfig.ResponseCacheConfig & {
     cache: KeyValueCache;
   },
-): Plugin;
+): GatewayPlugin;
 export default function useMeshResponseCache(
   options:
     | ResponseCacheConfig
     // TODO: This is for v0 compatibility, remove once v1 is released
     | (YamlConfig.ResponseCacheConfig & {
         cache: KeyValueCache;
       }),
-): Plugin {
-  const ttlPerType: Record<string, number> = { ...(options as ResponseCacheConfig).ttlPerType };
-  const ttlPerSchemaCoordinate: Record<string, number> = {
-    ...(options as ResponseCacheConfig).ttlPerSchemaCoordinate,
-  };
+): GatewayPlugin {
+  const ttlPerType: Record<string, number> = {};
+  if ('ttlPerType' in options && options.ttlPerType) {
+    for (const type in options.ttlPerType) {
+      const ttl = options.ttlPerType[type];
+      ttlPerType[type] = ttl;
+    }
+  }
 
-  const { ttlPerCoordinate } = options as YamlConfig.ResponseCacheConfig;
-  if (ttlPerCoordinate) {
-    for (const ttlConfig of ttlPerCoordinate) {
+  const ttlPerSchemaCoordinate: Record<string, number> = {};
+
+  if ('ttlPerSchemaCoordinate' in options && options.ttlPerSchemaCoordinate) {
+    for (const coordinate in options.ttlPerSchemaCoordinate) {
+      const ttl = options.ttlPerSchemaCoordinate[coordinate];
+      ttlPerSchemaCoordinate[coordinate] = ttl;
+    }
+  }
+
+  if ('ttlPerCoordinate' in options && options.ttlPerCoordinate?.length) {
+    for (const ttlConfig of options.ttlPerCoordinate) {
       ttlPerSchemaCoordinate[ttlConfig.coordinate] = ttlConfig.ttl;
     }
   }
```

Comment:
Why did you have to make this change ? Not sure to see difference here.

Reply:
To avoid `as`, and better benefit from function overriding above

Reply:
The as trick was there only because of Yaml retro-compatibility.

Not sure to understand what you mean by function overriding

Reply:
`as` overrides the type checking there completely. 
If you break the function overloading in the original signature, with `as` you will never see the potential errors.
If you avoid `as` but use the property check with `in`, you don't need to cast it explicitly to one single type, and will still be able to see potential problems just in case the function signature above is broken.

Reply:
Ah ok I understand what you mean.
You can keep it like that. I just find it very verbose for a very basic task ^^'

Reply:
I just saw it and looked unsafe for a future deprecation of Mesh v0.

Reply:
Anyway removed that diff 👍